### PR TITLE
[KEYCLOAK-18819] - SecureResponseType executor shall allow response_type=code when using JARM and response_mode=jwt

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

File: services/src/main/java/org/keycloak/services/clientpolicy/executor/SecureResponseTypeExecutor.java [hunk 1]
```diff
@@ -128,6 +129,15 @@ public void executeOnAuthorizationRequest(
             }
         }
 
+        if (request.getResponseMode() != null) {
+            if (parsedResponseType.hasResponseType(OIDCResponseType.CODE)) {
```
Comment (mposolda) [inline services/src/main/java/org/keycloak/services/clientpolicy/executor/SecureResponseTypeExecutor.java:133]: I think that this condition will be true also for the case when for example `response_type=code id_token` . However if I understand correctly the specification https://openid.net/specs/openid-financial-api-part-2-1_0.html#authorization-server , it mentions that:
```
the response_type value code in conjunction with the response_mode value jwt; 
```

So if I understand correctly, the response_type must be exactly `code` , which means that ID Token (or access token) should not be sent in the authorization response when JARM is used.

Does it makes sense to you? If yes, will it be possible to slightly update this and make sure we have test for the case when `response_type=code id_token` is rejected when `response_mode=jwt` is used?

**Review by mposolda — CHANGES_REQUESTED** [review]
@pedroigor Added one inline comment. Does it makes sense to you?

File: services/src/main/java/org/keycloak/services/clientpolicy/executor/SecureResponseTypeExecutor.java (hunk 1 above)
Comment (pedroigor) [reply services/src/main/java/org/keycloak/services/clientpolicy/executor/SecureResponseTypeExecutor.java:133]: The previous check already checks if the hybrid flow was requested. Pretty much the same behavior as before.

But if we want to be more explicit and expect a single response type in the list of response types, we can do it. I think the specs are basically protecting code using JARM so for me a `has` is OK.

Whatever you prefer.

**Review by mposolda — DISMISSED** [review]
@pedroigor Thanks! I am approving (hopefully test will pass).

**Comment by pedroigor** [comment]
@mposolda I noticed that we previously did not have a test when allowing token response type is requested and the policy is set to not allow. I've added one.

**Review by mposolda — APPROVED** [review]
@pedroigor Thanks for the update and adding the test!

**Pull request merged**